app/handlers/handlers_resource.py

```diff
@@ -3,20 +3,27 @@
 import kopf
 
 from app.api import TwingateAPIClient
-from app.crds import K8sMetadata, ResourceSpec
+from app.crds import ResourceSpec
 from app.handlers.base import fail, success
 
 
 @kopf.on.create("twingateresource")
-def twingate_resource_create(body, meta, spec, memo, logger, patch, **kwargs):
-    logger.info("Got a create request: %s. Metadata: %s", spec, meta)
-    k8s_metadata = K8sMetadata(**meta)
+def twingate_resource_create(body, labels, spec, memo, logger, patch, **kwargs):
+    logger.info("Got a create request: %s. Labels: %s", spec, labels)
     resource = ResourceSpec(**spec)
     client = TwingateAPIClient(memo.twingate_settings, logger=logger)
+    labels = {
+        **{
+            tag["name"]: tag["value"]
+            for tag in memo.twingate_settings.default_resource_tags
+        },
+        **dict(labels),
+    }
+    graphql_arguments = resource.to_graphql_arguments(labels=labels)
 
     # Support importing existing resources - if `id` already exist we assume it's already created
     if resource.id:
-        resource = client.resource_update(resource, k8s_metadata)
+        resource = client.resource_update(**graphql_arguments)
```

Comment:
Another alternative we discussed was to use [`client_resource.Resource`](https://github.com/Twingate/kubernetes-operator/blob/9fb816874e298b42c7b574f734bb800a05fabc29/app/api/client_resources.py#L66-L82), which represents a Twingate Resource. This is probably not a good idea because we don't have all the fields like `id`, `created_at` to construct the model instance.

Another option is to create a `ResourceArguments` pydantic model with fields as the Resource graphQL mutation arguments. Then we convert `ResourceSpec` to `ResourceArguments`. I think this option is overkill and would end up being very similar to the current implementation.